src/install/install.zig

```diff
@@ -911,66 +911,74 @@ const PackageInstall = struct {
         };
     }
 
-    fn verifyPackageJSONNameAndVersion(this: *PackageInstall) bool {
-        var allocator = this.allocator;
-        var total: usize = 0;
-        var read: usize = 0;
-
-        bun.copy(u8, this.destination_dir_subpath_buf[this.destination_dir_subpath.len..], std.fs.path.sep_str ++ "package.json");
-        this.destination_dir_subpath_buf[this.destination_dir_subpath.len + std.fs.path.sep_str.len + "package.json".len] = 0;
-        var package_json_path: [:0]u8 = this.destination_dir_subpath_buf[0 .. this.destination_dir_subpath.len + std.fs.path.sep_str.len + "package.json".len :0];
-        defer this.destination_dir_subpath_buf[this.destination_dir_subpath.len] = 0;
-
-        var package_json_file = this.destination_dir.dir.openFileZ(package_json_path, .{ .mode = .read_only }) catch return false;
-        defer package_json_file.close();
+    const IncrementalEqlsChecker = struct {
+        const Self = @This();
+        chars_to_match: string,
+        failed: bool = false,
 
-        var body_pool = Npm.Registry.BodyPool.get(allocator);
-        var mutable: MutableString = body_pool.data;
-        defer {
-            body_pool.data = mutable;
-            Npm.Registry.BodyPool.release(body_pool);
+        pub fn check(self: *Self, name: string) void {
+            if (self.failed) return;
+            if (name.len > self.chars_to_match.len) {
+                self.failed = true;
+                return;
+            }
+            for (name, self.chars_to_match[0..name.len]) |c1, c2| {
+                if (c1 != c2) {
+                    self.failed = true;
+                    return;
+                }
+            }
+            self.chars_to_match = self.chars_to_match[name.len..];
         }
 
-        mutable.reset();
-        mutable.list.expandToCapacity();
-
-        // Heuristic: most package.jsons will be less than 2048 bytes.
-        read = package_json_file.read(mutable.list.items[total..]) catch return false;
-        var remain = mutable.list.items[@min(total, read)..];
-        if (read > 0 and remain.len < 1024) {
-            mutable.growBy(4096) catch return false;
-            mutable.list.expandToCapacity();
```

Comment:
This is the logic I was talking about early this week which I believe makes no sense at all. That `@min` function always returns `0`, and `remain.len` is always larger than `1024` (actually, it is always `2048` and up). It also shouldn't assume that `growBy(4096)` is necessary just because there is space left. In fact, it is the opposite unless you hit the limit for how much can be read at once, which should be exceptionally rare.